portal-bridge/src/beacon_bridge.rs

```diff
@@ -200,34 +215,26 @@ impl BeaconBridge {
                     })?,
             });
 
-            // Update the latest finalized block root if we successfully gossiped the latest bootstrap.
-            match Self::gossip_beacon_content(portal_clients, content_key, content_value).await {
-                Ok(_) => {
-                    let mut finalized_block_root = finalized_block_root
-                        .lock()
-                        .expect("Unable to lock finalized block root");
-                    *finalized_block_root = latest_finalized_block_root;
-                }
-                Err(err) => return Err(err),
-            }
+            // Return the latest finalized block root if we successfully gossiped the latest bootstrap.
+            Self::gossip_beacon_content(portal_clients, content_key, content_value)
+                .await
+                .map(|_| latest_finalized_block_root)
+        } else {
+            Err(anyhow::anyhow!("Finalized block root has not changed"))
```

Comment:
We have to return Ok() here because it is expected behavior for the latest finalized block root to not change every slot:

```suggestion
            Ok(latest_finalized_block_root)
```